sklearn/cross_decomposition/_pls.py

```diff
@@ -221,31 +221,52 @@ def fit(self, X, Y):
             Training vectors, where `n_samples` is the number of samples and
             `n_features` is the number of predictors.
 
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target vectors, where `n_samples` is the number of samples and
+            `n_targets` is the number of response variables.
+
         Y : array-like of shape (n_samples,) or (n_samples, n_targets)
             Target vectors, where `n_samples` is the number of samples and
             `n_targets` is the number of response variables.
 
+            .. deprecated:: 1.5
+               `Y` is deprecated in 1.5 and will be removed in 1.7. Use `y` instead.
+
         Returns
         -------
         self : object
             Fitted model.
         """
-        check_consistent_length(X, Y)
+        if Y is not None:
+            warnings.warn(
+                "`Y` is deprecated in 1.5 and will be removed in 1.7. Use `y` instead.",
+                FutureWarning,
+            )
+            if y is not None:
+                raise ValueError(
+                    "Cannot use both `y` and `Y`. Use only `y` as `Y` is deprecated."
+                )
+            y = Y
```

Comment:
Maybe you could factor this into a separate function since you're reusing it in many places.